### PR TITLE
MB-11478: Service item calculations use ReferenceDate, label change for NTS-release

### PR DESCRIPTION
**Pull request by kylehilltruss**
## Summary

This pull request changes all uses of the `RequestedPickupDate` named parameter in service item calculations to use `ReferenceDate` instead. (It also adds `ReferenceDate` as one of the service item named params in the swagger definitions; earlier back end work had added this as a potentially returnable field.) 

It also changes any applicable labels in the front end `ServiceItemCalculations` from "Requested pickup," to "Actual pickup," for NTS-release shipments.

New stories and tests are added accordingly.

@transcom/truss-design: The layout of elements in the UI should not have changed, aside from conditionally replacing label text generated by the back end code.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make office_client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. In Storybook, view the "DLH" and "DLH (NTS-release)" stories for the `ServiceItemCalculations` component. Verify that the differences ("Requested pickup" and "Actual pickup") look correct.
2. Logging into the application locally, Create a move (or use an existing one) with both HHG and an NTS-release shipment. Navigate through the Customer, Service Counselor, TOO, and Prime Simulator flows as normal in order to generate payment request(s) that contain at least the "Domestic Linehaul" service items.
3. Log in to the Office application as a TIO user that can approve those payment requests. Review weights as normal. 
4. Opening the payment request approval flow, view the service item calculations for the HHG and NTS-release shipments. Verify that the HHG shipment displays the correct requested pickup date, and the label displays "Requested pickup:". Verify that the NTS-release shipment displays the correct _actual_ pickup date, and the label displays "Actual pickup:".
5. After approving those payment request(s), expand the service item rows to view the calculations, and verify that the labels and values look correct there as well.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

## Screenshots

* [HHG shipment, unchanged](https://user-images.githubusercontent.com/83614364/156223980-d602270b-9e77-4fcc-a1f6-fb8dba8d3013.png)
* [NTS-release shipment, label changed and with accurate value](https://user-images.githubusercontent.com/83614364/156223986-262a697c-086f-4bb1-8878-00171a5745e2.png)

### CODE REVIEW

**Review by YanZ777 — APPROVED** [review]
I have some non blocking comments on the PR, but everything else looks good to me.

File: src/constants/serviceItems.js [hunk 1]
```diff
@@ -55,6 +56,7 @@ const SERVICE_ITEM_CALCULATION_LABELS = {
   // Domestic non-peak or Domestic peak
   [SERVICE_ITEM_PARAM_KEYS.IsPeak]: 'Domestic',
   [SERVICE_ITEM_PARAM_KEYS.OriginPrice]: 'Origin price',
+  [SERVICE_ITEM_PARAM_KEYS.ReferenceDate]: 'Requested pickup',
```
[the remark this answers is not in the document]
Comment (kylehilltruss) [reply src/constants/serviceItems.js:59]: Yep. This is the default label for this named param key, when used. (The `ReferenceDate` named param value is also used in a couple other places, but those places have a built-in hardcoded label.)

File: src/constants/serviceItems.js [hunk 2]
```diff
@@ -55,6 +56,7 @@ const SERVICE_ITEM_CALCULATION_LABELS = {
...
   [SERVICE_ITEM_PARAM_KEYS.RequestedPickupDate]: 'Requested pickup',
```
[the remark this answers is not in the document]
Comment (kylehilltruss) [reply src/constants/serviceItems.js:60]: (Did you mean to point at the same line as your previous comment here?)

File: swagger-def/definitions/ServiceItemParamName.yaml [hunk 3]
```diff
@@ -44,6 +44,7 @@ enum:
   - PSI_ShippingLinehaulIntlOOPrice
   - RateAreaNonStdDest
   - RateAreaNonStdOrigin
+  - ReferenceDate
```
Comment (reggieriser) [inline swagger-def/definitions/ServiceItemParamName.yaml:47]: Thanks for the catch on this and adding it on the backend!

**Review by reggieriser — APPROVED** [review]
Code looks good!  Just had one minor nit (optional to fix).  Did a test scenario with a move with an HHG, NTS, and NTS-Release, then priced DLH for each.  Saw the requested pickup date label for the first two and the actual pickup date label for the last one, as expected. 🚢

File: src/constants/serviceItems.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (YanZ777) [reply src/constants/serviceItems.js:60]: Nope! It looks like both the `ReferenceDate` and the `RequestedPickupDate` are needed here, so you can disregard the above comment.

**Review by brooksr4 — APPROVED** [review]
Reviewed in storybook. Looks good.

**Pull request merged**